### DISCUSSION TITLE
include directive for markdown files

### DISCUSSION THREAD

**Discussion by mluerig** [executablebooks/meta#1102] [Q&A] [answered]
I thought with the below snippet I could include a markdown file in a jupyter notebook, but it doesn't work - other directives like figure or code work. am I doing something wrong or is this not supported?

```
    ```{include} ../_assets/md/my-file.md

    ```
```

**Comment by NickleDave** [comment]
Hi @mluerig :thinking: I think this should work? Not sure what the details of your setup are.

Does any of the info here help?
https://myst-parser.readthedocs.io/en/latest/faq/index.html#include-rst-files-into-a-markdown-file

**Reply by mluerig** [reply]
I have done what is shown in the FAQ.  Setup-wise I have msyt-nb installed, as well as jupyterlab-myst. also, the image directive works as expected in the same notebook 

    # packages in environment at C:\miniforge3\envs\pp-dev:
    #
    # Name                    Version                   Build  Channel
    alabaster                 0.7.13             pyhd8ed1ab_0    conda-forge
    anyio                     4.0.0                    pypi_0    pypi
    argon2-cffi               23.1.0                   pypi_0    pypi
    argon2-cffi-bindings      21.2.0                   pypi_0    pypi
    arrow                     1.3.0              pyhd8ed1ab_0    conda-forge
    astroid                   2.15.8           py39hcbf5309_0    conda-forge
    asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge
    async-lru                 2.0.4                    pypi_0    pypi
    atomicwrites              1.4.1              pyhd8ed1ab_0    conda-forge
    attrs                     23.1.0             pyh71513ae_1    conda-forge
    autopep8                  2.0.4              pyhd8ed1ab_0    conda-forge
    babel                     2.13.1             pyhd8ed1ab_0    conda-forge
    backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
    backports                 1.0                pyhd8ed1ab_3    conda-forge
    backports.functools_lru_cache 1.6.5              pyhd8ed1ab_0    conda-forge
    bcrypt                    4.0.1            py39hf21820d_1    conda-forge
    beautifulsoup4            4.12.2             pyha770c72_0    conda-forge
    binaryornot               0.4.4                      py_1    conda-forge
    black                     23.10.1          py39hcbf5309_0    conda-forge
    bleach                    6.1.0              pyhd8ed1ab_0    conda-forge
    brotli-python             1.1.0            py39h99910a6_1    conda-forge
    bzip2                     1.0.8                h8ffe710_4    conda-forge
    ca-certificates           2023.7.22            h56e8100_0    conda-forge
    cachetools                5.3.2                    pypi_0    pypi
    certifi                   2023.7.22          pyhd8ed1ab_0    conda-forge
    cffi                      1.16.0           py39ha55989b_0    conda-forge
    chardet                   5.2.0            py39hcbf5309_1    conda-forge
    charset-normalizer        3.3.1              pyhd8ed1ab_0    conda-forge
    click                     8.1.7           win_pyh7428d3b_0    conda-forge
    clip                      1.0                      pypi_0    pypi
    cloudpickle               3.0.0              pyhd8ed1ab_0    conda-forge
    colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
    colour                    0.1.5                    pypi_0    pypi
    comm                      0.1.4              pyhd8ed1ab_0    conda-forge
    contourpy                 1.2.0                    pypi_0    pypi
    cookiecutter              2.4.0              pyhca7485f_0    conda-forge
    coverage                  7.3.2                    pypi_0    pypi
    cryptography              41.0.5           py39hb6bd5e6_0    conda-forge
    cycler                    0.12.1                   pypi_0    pypi
    debugpy                   1.8.0            py39h99910a6_1    conda-forge
    decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
    defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
    diff-match-patch          20230430           pyhd8ed1ab_0    conda-forge
    dill                      0.3.7              pyhd8ed1ab_0    conda-forge
    distlib                   0.3.7                    pypi_0    pypi
    docopt                    0.6.2                    pypi_0    pypi
    docstring-to-markdown     0.13               pyhd8ed1ab_0    conda-forge
    docutils                  0.19                     pypi_0    pypi
    entrypoints               0.4                pyhd8ed1ab_0    conda-forge
    exceptiongroup            1.1.3              pyhd8ed1ab_0    conda-forge
    executing                 2.0.1              pyhd8ed1ab_0    conda-forge
    exifread                  3.0.0                    pypi_0    pypi
    filelock                  3.13.1                   pypi_0    pypi
    flake8                    6.0.0              pyhd8ed1ab_0    conda-forge
    fonttools                 4.44.0                   pypi_0    pypi
    fqdn                      1.5.1                    pypi_0    pypi
    fsspec                    2023.4.0                 pypi_0    pypi
    ftfy                      6.1.1                    pypi_0    pypi
    furo                      2023.3.27                pypi_0    pypi
    gettext                   0.21.1               h5728263_0    conda-forge
    glib                      2.78.0               h12be248_0    conda-forge
    glib-tools                2.78.0               h12be248_0    conda-forge
    google-api-core           2.14.0                   pypi_0    pypi
    google-api-python-client  2.107.0                  pypi_0    pypi
    google-auth               2.23.4                   pypi_0    pypi
    google-auth-httplib2      0.1.1                    pypi_0    pypi
    googleapis-common-protos  1.61.0                   pypi_0    pypi
    greenlet                  3.0.1                    pypi_0    pypi
    gst-plugins-base          1.22.6               h001b923_2    conda-forge
    gstreamer                 1.22.6               hb4038d2_2    conda-forge
    httplib2                  0.22.0                   pypi_0    pypi
    icu                       73.2                 h63175ca_0    conda-forge
    idna                      3.4                pyhd8ed1ab_0    conda-forge
    imageio                   2.31.6                   pypi_0    pypi
    imagesize                 1.4.1              pyhd8ed1ab_0    conda-forge
    importlib-metadata        6.8.0              pyha770c72_0    conda-forge
    importlib_metadata        6.8.0                hd8ed1ab_0    conda-forge
    importlib_resources       6.1.0              pyhd8ed1ab_0    conda-forge
    inflection                0.5.1              pyh9f0ad1d_0    conda-forge
    iniconfig                 2.0.0                    pypi_0    pypi
    intervaltree              3.1.0              pyhd8ed1ab_1    conda-forge
    ipykernel                 6.26.0             pyha63f2e9_0    conda-forge
    ipyplot                   1.1.1                    pypi_0    pypi
    ipython                   8.16.1             pyh5737063_0    conda-forge
    ipython_genutils          0.2.0                      py_1    conda-forge
    ipywidgets                8.1.1                    pypi_0    pypi
    isoduration               20.11.0                  pypi_0    pypi
    isort                     5.12.0             pyhd8ed1ab_1    conda-forge
    jaraco.classes            3.3.0              pyhd8ed1ab_0    conda-forge
    jedi                      0.18.2             pyhd8ed1ab_0    conda-forge
    jellyfish                 1.0.1            py39hf21820d_1    conda-forge
    jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
    json5                     0.9.14                   pypi_0    pypi
    jsonpointer               2.4                      pypi_0    pypi
    jsonschema                4.19.2             pyhd8ed1ab_0    conda-forge
    jsonschema-specifications 2023.7.1           pyhd8ed1ab_0    conda-forge
    jupyter                   1.0.0                    pypi_0    pypi
    jupyter-cache             0.6.1                    pypi_0    pypi
    jupyter-console           6.6.3                    pypi_0    pypi
    jupyter-events            0.8.0                    pypi_0    pypi
    jupyter-lsp               2.2.0                    pypi_0    pypi
    jupyter-server            2.9.1                    pypi_0    pypi
    jupyter-server-terminals  0.4.4                    pypi_0    pypi
    jupyter_client            8.5.0              pyhd8ed1ab_0    conda-forge
    jupyter_core              5.5.0            py39hcbf5309_0    conda-forge
    jupyterlab                4.0.7                    pypi_0    pypi
    jupyterlab-myst           2.1.0                    pypi_0    pypi
    jupyterlab-server         2.25.0                   pypi_0    pypi
    jupyterlab-widgets        3.0.9                    pypi_0    pypi
    jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
    keras                     2.15.0                   pypi_0    pypi
    keyring                   24.2.0           py39hcbf5309_1    conda-forge
    kiwisolver                1.4.5                    pypi_0    pypi
    krb5                      1.21.2               heb0366b_0    conda-forge
    lazy-object-proxy         1.9.0            py39ha55989b_1    conda-forge
    libclang                  15.0.7          default_h77d9078_3    conda-forge
    libclang13                15.0.7          default_h77d9078_3    conda-forge
    libffi                    3.4.2                h8ffe710_5    conda-forge
    libglib                   2.78.0               he8f3873_0    conda-forge
    libiconv                  1.17                 h8ffe710_0    conda-forge
    libjpeg-turbo             3.0.0                hcfcfb64_1    conda-forge
    libogg                    1.3.4                h8ffe710_1    conda-forge
    libpng                    1.6.39               h19919ed_0    conda-forge
    libsodium                 1.0.18               h8d14728_1    conda-forge
    libspatialindex           1.9.3                h39d44d4_4    conda-forge
    libsqlite                 3.43.2               hcfcfb64_0    conda-forge
    libvorbis                 1.3.7                h0e60522_0    conda-forge
    libwebp                   1.3.2                hcfcfb64_1    conda-forge
    libwebp-base              1.3.2                hcfcfb64_0    conda-forge
    libzlib                   1.2.13               hcfcfb64_5    conda-forge
    markdown-it-py            2.2.0                    pypi_0    pypi
    markupsafe                2.1.3            py39ha55989b_1    conda-forge
    matplotlib                3.8.1                    pypi_0    pypi
    matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
    mccabe                    0.7.0              pyhd8ed1ab_0    conda-forge
    mdit-py-plugins           0.3.5                    pypi_0    pypi
    mdurl                     0.1.0              pyhd8ed1ab_0    conda-forge
    mistune                   3.0.1              pyhd8ed1ab_0    conda-forge
    mock                      5.1.0                    pypi_0    pypi
    more-itertools            10.1.0             pyhd8ed1ab_0    conda-forge
    mpmath                    1.3.0                    pypi_0    pypi
    mypy_extensions           1.0.0              pyha770c72_0    conda-forge
    myst-nb                   0.17.2                   pypi_0    pypi
    myst-parser               0.18.1                   pypi_0    pypi
    nbclient                  0.7.4                    pypi_0    pypi
    nbconvert                 7.10.0             pyhd8ed1ab_0    conda-forge
    nbconvert-core            7.10.0             pyhd8ed1ab_0    conda-forge
    nbconvert-pandoc          7.10.0             pyhd8ed1ab_0    conda-forge
    nbformat                  5.9.2              pyhd8ed1ab_0    conda-forge
    nbsphinx                  0.9.3                    pypi_0    pypi
    nest-asyncio              1.5.8              pyhd8ed1ab_0    conda-forge
    networkx                  3.0                      pypi_0    pypi
    nh3                       0.2.14                   pypi_0    pypi
    notebook                  7.0.6                    pypi_0    pypi
    notebook-shim             0.2.3                    pypi_0    pypi
    numpy                     1.26.1                   pypi_0    pypi
    numpydoc                  1.5.0              pyhd8ed1ab_0    conda-forge
    oauth2client              4.1.3                    pypi_0    pypi
    opencv-contrib-python     4.5.2.54                 pypi_0    pypi
    opencv-python             4.8.1.78                 pypi_0    pypi
    openssl                   3.1.4                hcfcfb64_0    conda-forge
    overrides                 7.4.0                    pypi_0    pypi
    packaging                 23.2               pyhd8ed1ab_0    conda-forge
    pandas                    2.1.2                    pypi_0    pypi
    pandoc                    3.1.3                h57928b3_0    conda-forge
    pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
    paramiko                  3.3.1              pyhd8ed1ab_0    conda-forge
    parso                     0.8.3              pyhd8ed1ab_0    conda-forge
    pathspec                  0.11.2             pyhd8ed1ab_0    conda-forge
    pcre2                     10.40                h17e33f8_0    conda-forge
    pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
    phenopype                 4.0.0                     dev_0    <develop>
    pickleshare               0.7.5                   py_1003    conda-forge
    pillow                    10.1.0                   pypi_0    pypi
    pip                       23.3.1             pyhd8ed1ab_0    conda-forge
    pkginfo                   1.9.6                    pypi_0    pypi
    pkgutil-resolve-name      1.3.10             pyhd8ed1ab_1    conda-forge
    platformdirs              3.11.0             pyhd8ed1ab_0    conda-forge
    pluggy                    1.3.0              pyhd8ed1ab_0    conda-forge
    ply                       3.11                       py_1    conda-forge
    prometheus-client         0.18.0                   pypi_0    pypi
    prompt-toolkit            3.0.39             pyha770c72_0    conda-forge
    prompt_toolkit            3.0.39               hd8ed1ab_0    conda-forge
    protobuf                  4.25.0                   pypi_0    pypi
    psutil                    5.9.5            py39ha55989b_1    conda-forge
    ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
    pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
    pyasn1                    0.5.0                    pypi_0    pypi
    pyasn1-modules            0.3.0                    pypi_0    pypi
    pycodestyle               2.10.0             pyhd8ed1ab_0    conda-forge
    pycparser                 2.21               pyhd8ed1ab_0    conda-forge
    pydocstyle                6.3.0              pyhd8ed1ab_0    conda-forge
    pydrive                   1.3.1                    pypi_0    pypi
    pyflakes                  3.0.1              pyhd8ed1ab_0    conda-forge
    pygments                  2.16.1             pyhd8ed1ab_0    conda-forge
    pykwalify                 1.8.0                    pypi_0    pypi
    pylint                    2.17.7             pyhd8ed1ab_0    conda-forge
    pylint-venv               3.0.3              pyhd8ed1ab_0    conda-forge
    pyls-spyder               0.4.0              pyhd8ed1ab_0    conda-forge
    pynacl                    1.5.0            py39h09fa780_3    conda-forge
    pyparsing                 3.1.1                    pypi_0    pypi
    pyproject-api             1.6.1                    pypi_0    pypi
    pyqt                      5.15.9           py39hb77abff_5    conda-forge
    pyqt5-sip                 12.12.2          py39h99910a6_5    conda-forge
    pyqtwebengine             5.15.9           py39h2f4a3f1_5    conda-forge
    pyradiomics               3.0.1                    pypi_0    pypi
    pysocks                   1.7.1              pyh0701188_6    conda-forge
    pytest                    7.4.3                    pypi_0    pypi
    pytest-cov                4.1.0                    pypi_0    pypi
    pytest-xvfb               3.0.0                    pypi_0    pypi
    python                    3.9.18          h4de0772_0_cpython    conda-forge
    python-coveralls          2.9.3                    pypi_0    pypi
    python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
    python-fastjsonschema     2.18.1             pyhd8ed1ab_0    conda-forge
    python-json-logger        2.0.7                    pypi_0    pypi
    python-lsp-black          1.3.0              pyhd8ed1ab_0    conda-forge
    python-lsp-jsonrpc        1.1.2              pyhd8ed1ab_0    conda-forge
    python-lsp-server         1.7.4              pyhd8ed1ab_0    conda-forge
    python-lsp-server-base    1.7.4              pyhd8ed1ab_0    conda-forge
    python-slugify            8.0.1              pyhd8ed1ab_2    conda-forge
    python_abi                3.9                      4_cp39    conda-forge
    pytoolconfig              1.2.5              pyhd8ed1ab_0    conda-forge
    pytz                      2023.3.post1       pyhd8ed1ab_0    conda-forge
    pyvirtualdisplay          3.0                      pypi_0    pypi
    pywavelets                1.4.1                    pypi_0    pypi
    pywin32                   306              py39h99910a6_2    conda-forge
    pywin32-ctypes            0.2.2            py39hcbf5309_1    conda-forge
    pywinpty                  2.0.12                   pypi_0    pypi
    pyyaml                    6.0.1            py39ha55989b_1    conda-forge
    pyzmq                     25.1.1           py39h7eaf5a6_2    conda-forge
    qdarkstyle                3.1                pyhd8ed1ab_0    conda-forge
    qstylizer                 0.2.2              pyhd8ed1ab_0    conda-forge
    qt-main                   5.15.8              h9e85ed6_17    conda-forge
    qt-webengine              5.15.8               h5b1ea0b_0    conda-forge
    qtawesome                 1.2.3              pyhd8ed1ab_0    conda-forge
    qtconsole                 5.4.4              pyhd8ed1ab_0    conda-forge
    qtconsole-base            5.4.4              pyha770c72_0    conda-forge
    qtpy                      2.4.1              pyhd8ed1ab_0    conda-forge
    rawpy                     0.18.1                   pypi_0    pypi
    readme-renderer           42.0                     pypi_0    pypi
    referencing               0.30.2             pyhd8ed1ab_0    conda-forge
    regex                     2023.10.3                pypi_0    pypi
    requests                  2.31.0             pyhd8ed1ab_0    conda-forge
    requests-toolbelt         1.0.0                    pypi_0    pypi
    rfc3339-validator         0.1.4                    pypi_0    pypi
    rfc3986                   2.0.0                    pypi_0    pypi
    rfc3986-validator         0.1.1                    pypi_0    pypi
    rich                      13.6.0             pyhd8ed1ab_0    conda-forge
    rope                      1.10.0             pyhd8ed1ab_0    conda-forge
    rpds-py                   0.10.6           py39hf21820d_0    conda-forge
    rsa                       4.9                      pypi_0    pypi
    rtree                     1.1.0            py39h09fdee3_0    conda-forge
    ruamel-yaml               0.16.12                  pypi_0    pypi
    scipy                     1.11.3                   pypi_0    pypi
    seaborn                   0.13.0                   pypi_0    pypi
    send2trash                1.8.2                    pypi_0    pypi
    setuptools                68.2.2             pyhd8ed1ab_0    conda-forge
    shortuuid                 1.0.11                   pypi_0    pypi
    simpleitk                 2.3.1                    pypi_0    pypi
    sip                       6.7.12           py39h99910a6_0    conda-forge
    six                       1.16.0             pyh6c4a22f_0    conda-forge
    sniffio                   1.3.0                    pypi_0    pypi
    snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
    sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
    soupsieve                 2.5                pyhd8ed1ab_1    conda-forge
    sphinx                    5.3.0                    pypi_0    pypi
    sphinx-copybutton         0.5.2                    pypi_0    pypi
    sphinx-design             0.5.0                    pypi_0    pypi
    sphinx-multibuild         1.2.1                    pypi_0    pypi
    sphinxcontrib-applehelp   1.0.7              pyhd8ed1ab_0    conda-forge
    sphinxcontrib-devhelp     1.0.5              pyhd8ed1ab_0    conda-forge
    sphinxcontrib-htmlhelp    2.0.4              pyhd8ed1ab_0    conda-forge
    sphinxcontrib-jsmath      1.0.1              pyhd8ed1ab_0    conda-forge
    sphinxcontrib-qthelp      1.0.6              pyhd8ed1ab_0    conda-forge
    sphinxcontrib-serializinghtml 1.1.9              pyhd8ed1ab_0    conda-forge
    sphinxemoji               0.2.0                    pypi_0    pypi
    spyder                    5.4.5            py39hcbf5309_0    conda-forge
    spyder-kernels            2.4.4           win_pyh7428d3b_0    conda-forge
    sqlalchemy                2.0.22                   pypi_0    pypi
    stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
    sympy                     1.12                     pypi_0    pypi
    tabulate                  0.9.0                    pypi_0    pypi
    terminado                 0.17.1                   pypi_0    pypi
    text-unidecode            1.3                pyhd8ed1ab_1    conda-forge
    textdistance              4.5.0              pyhd8ed1ab_0    conda-forge
    three-merge               0.1.1              pyh9f0ad1d_0    conda-forge
    tifffile                  2023.9.26                pypi_0    pypi
    tinycss2                  1.2.1              pyhd8ed1ab_0    conda-forge
    tk                        8.6.13               hcfcfb64_0    conda-forge
    toml                      0.10.2             pyhd8ed1ab_0    conda-forge
    tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
    tomlkit                   0.12.1             pyha770c72_0    conda-forge
    torch                     2.1.0+cu121              pypi_0    pypi
    torchvision               0.16.0+cu121             pypi_0    pypi
    tornado                   6.3.3            py39ha55989b_1    conda-forge
    tox                       4.11.3                   pypi_0    pypi
    tqdm                      4.66.1                   pypi_0    pypi
    traitlets                 5.13.0             pyhd8ed1ab_0    conda-forge
    twine                     4.0.2                    pypi_0    pypi
    types-python-dateutil     2.8.19.14          pyhd8ed1ab_0    conda-forge
    typing-extensions         4.8.0                hd8ed1ab_0    conda-forge
    typing_extensions         4.8.0              pyha770c72_0    conda-forge
    tzdata                    2023.3                   pypi_0    pypi
    ucrt                      10.0.22621.0         h57928b3_0    conda-forge
    ujson                     5.8.0            py39h99910a6_0    conda-forge
    ultralytics               8.0.120                  pypi_0    pypi
    uri-template              1.3.0                    pypi_0    pypi
    uritemplate               4.1.1                    pypi_0    pypi
    urllib3                   2.0.7              pyhd8ed1ab_0    conda-forge
    vc                        14.3                h64f974e_17    conda-forge
    vc14_runtime              14.36.32532         hdcecf7f_17    conda-forge
    virtualenv                20.24.6                  pypi_0    pypi
    vs2015_runtime            14.36.32532         h05e6639_17    conda-forge
    watchdog                  3.0.0            py39hcbf5309_1    conda-forge
    wcwidth                   0.2.9              pyhd8ed1ab_0    conda-forge
    webcolors                 1.13                     pypi_0    pypi
    webencodings              0.5.1              pyhd8ed1ab_2    conda-forge
    websocket-client          1.6.4                    pypi_0    pypi
    whatthepatch              1.0.5              pyhd8ed1ab_0    conda-forge
    wheel                     0.41.3             pyhd8ed1ab_0    conda-forge
    widgetsnbextension        4.0.9                    pypi_0    pypi
    win_inet_pton             1.1.0              pyhd8ed1ab_6    conda-forge
    wrapt                     1.15.0           py39ha55989b_1    conda-forge
    xz                        5.2.6                h8d14728_0    conda-forge
    yaml                      0.2.5                h8ffe710_2    conda-forge
    yapf                      0.40.1             pyhd8ed1ab_0    conda-forge
    zeromq                    4.3.5                h63175ca_0    conda-forge
    zipp                      3.17.0             pyhd8ed1ab_0    conda-forge
    zstd                      1.5.5                h12be248_0    conda-forge

**Reply by rowanc1** [reply, marked as answer]
Note that the `include`/`literal-include` directives are not yet working in the jupyterlab-myst preview! We are tracking this improvement here:
https://github.com/executablebooks/jupyterlab-myst/issues/189

**Comment by mluerig** [comment]
ah ok - thanks for letting me know.
